sopel/config/types.py

```diff
@@ -216,31 +219,43 @@ def __get__(self, instance, owner=None):
             # (and is really just more intuitive) to return the descriptor
             # instance here.
             return self
-
+        value = None
         env_name = 'SOPEL_%s_%s' % (instance._section_name.upper(), self.name.upper())
         if env_name in os.environ:
             value = os.environ.get(env_name)
         elif instance._parser.has_option(instance._section_name, self.name):
             value = instance._parser.get(instance._section_name, self.name)
-        else:
-            if self.default is not NO_DEFAULT:
-                return self.default
-            raise AttributeError(
-                "Missing required value for {}.{}".format(
-                    instance._section_name, self.name
-                )
+
+        settings = instance._parent
+        section = getattr(settings, instance._section_name)
+        return self._parse(value, settings, section)
+
+    def _parse(self, value, settings, section):
+        if value is not None:
+            return self.parse(value)
+        if self.default is not NO_DEFAULT:
+            return self.default
```

Comment:
No suggestion or anything here, just wondering if we ever talked about if and/or how `None` should be allowed as a setting value in its own right. If not, we probably should at some point. I guess we'd need a new empty class like `UnsetOption` to differentiate then.

Reply:
It's the underlying implementation (Python's ConfigParser) that's leaking up the abstraction layer: basically, setting an option to "None" means "unset".